.github/workflows/build.yml

```diff
@@ -0,0 +1,25 @@
+name: Test the Build
+
+on:
+  # test on PRs and double-test on merges to master, or manually.
+  pull_request:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+    - name: Build tools
+      run: |
+          python -m pip install --upgrade pip
+          pip install build
+    - name: Build
+      run: |
+          perl -pi -e 's/\${dataset_name}/sample-dataset/' setup.py
```

Comment:
IIUC, this is a find and replace? 

Just out of curiosity, why `perl -pi -e`? (I tried googling and found [this SO answer](https://stackoverflow.com/questions/28131528/what-does-exactly-perl-pi-e-do), but I still don't entirely understand the purpose.)